lib/services/local/version.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package local
```

Comment:
Sure, but we're not trying to make this bulletproof. You could make the same argument for the backend - "couldn't the user just remove the version entry from the backend to violate the upgrade guidance?"

Reply:
I think keeping it in local process storage would also not work in cases when auth runs in Kubernetes and pods recycle (which is every upgrade basically).

Reply:
Actually, ignore my previous comment, we use secrets for local storage in Kubernetes so it would work. In fact, it looks like @fspmarshall added something called [InitialLocalVersion](https://github.com/gravitational/teleport/blob/c44530d985cf07ad4fa849b30281d09fc6ca2813/lib/auth/state/state.go#L90) to process storage a few weeks ago in https://github.com/gravitational/teleport/pull/42189. Is this something we can piggy-back on, or add the version we need to record alongside it? @fspmarshall Is "InitialLocalVersion" supposed to be updated when teleport upgrades?

Reply:
@r0mant I've launched latest v17, so we keep initial version `"initial_local_version":"15.4.5"`
what about to extend this state object to add version history field as array to keep track all version launch (as usually migration logic does)

Reply:
I don't know if we need to keep the version history necessarily but adding another process state field for tracking the latest and current version (unless we can reuse initial_local_version) seems reasonable.